Adding documentation about explain in hybrid query and hybrid_score_explanation processor (#9053)
opensearch-project/documentation-website: _search-plugins/hybrid-search.md

        ]
    }
}
```

## Explainability in hybrid query
**Introduced 2.19**
{: .label .label-purple }

> The explainability feature helps you understand how scores are calculated, normalized, and combined in hybrid queries. When enabled, it provides detailed information about the scoring process for each search result. This includes the various score normalization techniques used (such as `min_max` or `rrf`), how different scores are combined, and the details of individual score calculations from sub-queries (like [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) for text-based queries). This comprehensive insight into the scoring process makes it easier to understand and optimize your hybrid query results. For more information about explain, see corresponding [API reference page]({{site.url}}{{site.baseurl}}/api-reference/explain). 

Reviewer: `This includes the various score normalization techniques used (such as `min_max` or `rrf`), how different scores are combined, and the details of individual score calculations from sub-queries (like [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) for text-based queries). `

Can this sentence be little less wordy? I found a hard time in going through it and understand it completely. Now as I already have all the context so I was able to understand. But for CX it is a very wordy sentence will make there attention loose. WDYT?

Author: I can make that sentence simpler and drop some of the details, something like:
```
The explainability feature provides detailed information about the scoring process for each search result. It reveals the score normalization techniques used, how different scores are combined, and the calculations for individual sub-query scores.
```

Author: After I looked closer I've changed the whole paragraph to a simpler form. Hope that addressed concerns

The explain API is an expensive operation in terms of both resources and time. On production clusters, we recommend using it sparingly for the purpose of troubleshooting.
{: .warning }

You can use `explain` with the complete hybrid query using following syntax for URL 
```json
GET <index>/_search?search_pipeline=<search_pipeline>&explain=true
POST <index>/_search?search_pipeline=<search_pipeline>&explain=true
```

To use explainability, you must configure the `hybrid_score_explanation` response processor in your search pipeline and include the explain=true parameter in your search request. For more information, see [hybrid_score_explanation]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/explanation-processor). 

> If `explain` is used with the individual document id like in the following example then result will have only details of the low level scoring, like [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) for text-based queries like `term` or `match`.

Author: I haven't added such sample here because it doesn't have any new stuff we've added for hybrid query. Because the response will look exactly same as for any generic query, I can put a link to the main page with `explain`, is has an example of such response https://opensearch.org/docs/latest/api-reference/explain/. WDYT @ryanbogan?

> If `explain` is used with the individual document id like in the following example then result will have only details of the low level scoring, like [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) for text-based queries like `term` or `match`.

Reviewer 2: @martin-gaievski Yeah I think that's the best approach.  You're right that we don't need a whole example for this page since it's the same as the query for non-hybrid.  I think putting the link to the main documentation is the right solution.

> If `explain` is used with the individual document id like in the following example then result will have only details of the low level scoring, like [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) for text-based queries like `term` or `match`.

Author: I've added a link to main `explain` page with example of response

        ]
    }
}
```

## Explain
**Introduced 2.19**
{: .label .label-purple }

You can provide the `explain` parameter to understand how scores are calculated, normalized, and combined in hybrid queries. When enabled, it provides detailed information about the scoring process for each search result. This includes revealing the score normalization techniques used, how different scores were combined, and the calculations for individual subquery scores. This comprehensive insight makes it easier to understand and optimize your hybrid query results. For more information about `explain`, see [Explain API]({{site.url}}{{site.baseurl}}/api-reference/explain/). 

> Explain is an expensive operation in terms of both resources and time. On production clusters, we recommend using it sparingly for the purpose of troubleshooting.

Reviewer 3: ```suggestion
`explain` is an expensive operation in terms of both resources and time. For production clusters, we recommend using it sparingly for the purpose of troubleshooting.
```

{: .warning }

> You can provide the `explain` parameter in a URL when running a complete hybrid query using following syntax:

Reviewer 3: ```suggestion
You can provide the `explain` parameter in a URL when running a complete hybrid query using the following syntax:
```

```json
GET <index>/_search?search_pipeline=<search_pipeline>&explain=true
POST <index>/_search?search_pipeline=<search_pipeline>&explain=true
```

To use the `explain` parameter, you must configure the `hybrid_score_explanation` response processor in your search pipeline. For more information, see [Hybrid score explanation processor]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/explanation-processor/). 

You can also use `explain` with the individual document ID:

```json
GET <index>/_explain/<id>
POST <index>/_explain/<id>
```

> In this case, the result will contain only low-level scoring details, for example, [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) scores for text-based queries such as `term` or `match`. For an example response, see [Explain API example response]({{site.url}}{{site.baseurl}}/api-reference/explain/#example-response).

Reviewer 3: ```suggestion
In this case, the result will contain only low-level scoring information, for example, [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) scores for text-based queries such as `term` or `match`. For an example response, see [Explain API example response]({{site.url}}{{site.baseurl}}/api-reference/explain/#example-response).
```

> To see the explain output for all results, set the `explain` parameter to `true` either in the URL or in the request body:

Reviewer 3: ```suggestion
To see the `explain` output for all results, set the parameter to `true` either in the URL or in the request body:
```

```json
POST my-nlp-index/_search?search_pipeline=my_pipeline&explain=true
{
  "_source": {
    "exclude": [
      "passage_embedding"
    ]
  },
  "query": {
    "hybrid": {
      "queries": [
        {
          "match": {
            "text": {
              "query": "horse"
            }
          }
        },
        {
          "neural": {
            "passage_embedding": {
              "query_text": "wild west",
              "model_id": "aVeif4oB5Vm0Tdw8zYO2",
              "k": 5
            }
          }
        }
      ]
    }
  }
}
```
{% include copy-curl.html %}

> The response contains scoring explanation:

Reviewer 3: ```suggestion
The response contains scoring information:
```

                                                            "description": "avgdl, average length of field",
                                                            "details": []
                                                        }
                                                    ]
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            "value": 0.8503647,
                            "description": "min_max normalization of:",
                            "details": [
                                {
                                    "value": 0.015177966,
                                    "description": "within top 5",
                                    "details": []
                                }
                            ]
                        }
                    ]
...
```
</details>

### Response body fields

Field | Description
:--- | :---

> `explanation` | The `explanation` object has three properties: `value`, `description`, and `details`. The `value` shows the result of the calculation, the `description` explains what type of calculation was performed, and the `details` shows any subcalculations performed. For score normalization, the information in the `description` inludes the technique used for normalization or combination and the corresponding score. 

Reviewer 3: ```suggestion
`explanation` | The `explanation` object has three properties: `value`, `description`, and `details`. The `value` property shows the result of the calculation, `description` explains what type of calculation was performed, and `details` shows any subcalculations performed. For score normalization, the information in the `description` property includes the technique used for normalization or combination and the corresponding score. 
```
